src/target.js

```diff
@@ -200,24 +200,36 @@ export default class Target {
     }
   }
 
-  async toMatchSnapshot(image, callback) {
+  async toMatchSnapshot(image, callback, settings) {
```

Comment:
I wonder if we could move all these sanitizations to a separate function like `sanitiseToMatchSnapshotArg()`?!